src/logical/program_analysis/analysis.rs

```diff
@@ -467,6 +475,60 @@ impl Program {
         Ok(())
     }
 
+    // Check if there is a constant that cannot be converted into the type of
+    // the variable/predicate position it is compared to.
```

Comment:
```suggestion
    /// Check if there is a constant that cannot be converted into the type of
    /// the variable/predicate position it is compared to.
```
Use doc-comments instead